plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/delete/DeleteProjectPresenter.java

```diff
@@ -0,0 +1,179 @@
+import static org.eclipse.che.ide.ext.openshift.shared.OpenshiftProjectTypeConstants.OPENSHIFT_APPLICATION_VARIABLE_NAME;
+import static org.eclipse.che.ide.ext.openshift.shared.OpenshiftProjectTypeConstants.OPENSHIFT_NAMESPACE_VARIABLE_NAME;
+import static org.eclipse.che.ide.ext.openshift.shared.OpenshiftProjectTypeConstants.OPENSHIFT_PROJECT_TYPE_ID;
+
+/**
+ * @author Alexander Andrienko
+ */
+@Singleton
+public class DeleteProjectPresenter extends ValidateAuthenticationPresenter {
+    
+    private final AppContext                    appContext;
+    private final DialogFactory                 dialogFactory;
+    private final OpenshiftLocalizationConstant locale;
+    private final OpenshiftServiceClient        service;
+    private final NotificationManager           notificationManager;
+    private final ProjectServiceClient          projectService;
+    private final DtoUnmarshallerFactory        dtoUnmarshaller;
+
+    @Inject
+    protected DeleteProjectPresenter(OpenshiftAuthenticator openshiftAuthenticator,
+                                     OpenshiftAuthorizationHandler openshiftAuthorizationHandler,
+                                     AppContext appContext,
+                                     DialogFactory dialogFactory,
+                                     OpenshiftLocalizationConstant locale,
+                                     OpenshiftServiceClient service,
+                                     ProjectServiceClient projectService,
+                                     NotificationManager notificationManager,
+                                     DtoUnmarshallerFactory dtoUnmarshaller) {
+        super(openshiftAuthenticator, openshiftAuthorizationHandler, locale, notificationManager);
+
+        this.appContext = appContext;
+        this.dialogFactory = dialogFactory;
+        this.locale = locale;
+        this.service = service;
+        this.notificationManager = notificationManager;
+        this.projectService = projectService;
+        this.dtoUnmarshaller = dtoUnmarshaller;
+    }
+
+    @Override
+    protected void onSuccessAuthentication() {
+        ProjectDescriptor descriptor = appContext.getCurrentProject().getProjectDescription();
+        final String namespace = getAttributeValue(descriptor, OPENSHIFT_NAMESPACE_VARIABLE_NAME);
+        
+        if (!namespace.isEmpty()) {
+            Promise<List<BuildConfig>> buildConfigs = service.getBuildConfigs(namespace);
+            buildConfigs.then(showConfirmDialog(descriptor, namespace))
+                        .catchError(handleError(namespace));
+        } else {
+            notificationManager.showError(locale.projectAttributeNameSpaceIsEmpty());
+        }
+    }
+
+
+    private String getAttributeValue(ProjectDescriptor projectDescriptor, String value) {
+        List<String> attributes = projectDescriptor.getAttributes().get(value);
+        if (attributes == null || attributes.isEmpty()) {
+            return "";
```

Comment:
Why do you prefer return empty string instead of null value like https://github.com/codenvy/che-plugins/blob/openshift/plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/webhooks/ShowWebhooksPresenter.java#L125
I think null is more logic for place where this method will be called.